### PR TITLE
update doc about mongo and sth indexes by DM

### PR DESCRIPTION
**Pull request by AlvaroVega**
https://github.com/telefonicaid/fiware-cygnus/issues/2317

### CODE REVIEW

**Comment by AlvaroVega** [comment]
and maybe something about indexing procedure (re indexing) it would be nice

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_mongo_sink.md [hunk 1]
```diff
@@ -286,6 +287,25 @@ If `data_model=dm-by-entity` and `attr_persistence=column` then `NGSIMongoSink`
 
 [Top](#top)
 
+### <a name="section1.4"></a>MongoDB Indexes
+
+Since version 3.0.0 of cygnus index named `cyg_agg_opt` is created in this way but depending on datamodel:
```
Comment (fgalan) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_mongo_sink.md:292]: Should both indexed (cyg_agg_opt and cyg_raw_opt) be mentioned in both sinks (ngsi_mongo_sink.md and ngsi_sth_sink.md)?

As far as I remember (although I may be wrong) each index is used by one sink but I don't remember if it is this way

* mongo sink -> cyg_agg_opt
* sth sink -> cyg_raw_opt

or

* mongo sink -> cyg_raw_opt
* sth sink -> cyg_agg_opt

Comment (AlvaroVega) [reply doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_mongo_sink.md:292]: > Should both indexed (cyg_agg_opt and cyg_raw_opt) be mentioned in both sinks (ngsi_mongo_sink.md and ngsi_sth_sink.md)?
> 
> As far as I remember (although I may be wrong) each index is used by one sink but I don't remember if it is this way
> 
>     * mongo sink -> cyg_agg_opt
> 
>     * sth sink -> cyg_raw_opt
> 
> 
> or
> 
>     * mongo sink -> cyg_raw_opt
> 
>     * sth sink -> cyg_agg_opt

@fgalan you are rigth!. MongoSink creates Raw table and STHSInk creates Aggregate table.

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_mongo_sink.md [hunk 2]
```diff
@@ -286,6 +287,16 @@ If `data_model=dm-by-entity` and `attr_persistence=column` then `NGSIMongoSink`
...
 [Top](#top)
 
+### <a name="section1.4"></a>MongoDB Indexes
+Since version 3.0.0 of cygnus an index named `cyg_raw_opt` is created in this way but depending on datamodel:
+
+| datamodel       | keys                                                             |
+| :-------------- | :--------------------------------------------------------------- |
+| DMBYSERVICEPATH | recvTime, entityId, entityType, attrName, attrType, attrValue    |
+| DMBYENTITY      | recvTime, attrName, attrType, attrValue                          |
+| DMBYATTRIBUTE   | recvTime, attrType, attrValue                                    |
```
Comment (fgalan) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_mongo_sink.md:297]: Maybe we should use the names that are used in configuration by users (i.e `dm-by-...`). I understand that the `DM...` being used are the ones from the code (so, somehow "hidden" by regular users).

In addition, what about the other DMs (from https://github.com/telefonicaid/fiware-cygnus/blob/master/doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_postgresql_sink.md#section1.2 it seems we have 11 DMs, but this table only includes 3)? If they don't involve indexes maybe we should add a sentence like this:

> In the case of other datamodels than the ones included in the table above (including the default behaviour when no datamodel is set) cygnus doesn't create any index.

(Same applies to the modification done in PR https://github.com/telefonicaid/fiware-sth-comet/pull/624. Maybe a follow up PR in that repo should be done)

Comment (fgalan) [reply doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_mongo_sink.md]: Thus, I'd suggest to add the following note just after the table in documentaiton:

> Note that attempting to use any other datamodel different from these ones will cause a "Unknown data model" ERROR in log traces. In addition, note that default datamodel for STH/Mongo sink is ...

completing the missing "..."

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_sth_sink.md [hunk 3]
```diff
@@ -288,6 +289,19 @@ Assuming `data_model=dm-by-entity` and all the possible resolutions as configura
...
+Since version 3.0.0 of cygnus index named `cyg_agg_opt` is created in this way but depending on datamodel:
+
+| datamodel          | keys                                               |
+| :----------------- | :------------------------------------------------- |
+| dm-by-service-path | entityId, entityType, attrName, resolution, origin |
+| dm-by-entity       | attrName, resolution, origin                       |
+| dm-by-attribute    | resolution, origin                                 |
+
+
+Note that attempting to use any other datamodel different from these ones will cause a "Unknown data model" ERROR in log traces. In addition, note that default datamodel for STH/Mongo sink is dm-by-entity as all others sinks which uses datamodel.
```
Comment (fgalan) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_sth_sink.md:303]:
```suggestion
Note that attempting to use any other datamodel different from these ones will cause a "Unknown data model" ERROR in log traces. In addition, note that default datamodel for NGSISTHSink is dm-by-entity as all others sinks which uses datamodel.
```

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_mongo_sink.md [hunk 4]
```diff
@@ -286,6 +287,17 @@ If `data_model=dm-by-entity` and `attr_persistence=column` then `NGSIMongoSink`
...
+### <a name="section1.4"></a>MongoDB Indexes
+Since version 3.0.0 of cygnus an index named `cyg_raw_opt` is created in this way but depending on datamodel:
+
+| datamodel         | keys                                                             |
+| :---------------- | :--------------------------------------------------------------- |
+| dm-by-servicepath | recvTime, entityId, entityType, attrName, attrType, attrValue    |
+| dm-by-entity      | recvTime, attrName, attrType, attrValue                          |
+| dm-by-attribute   | recvTime, attrType, attrValue                                    |
+
+Note that attempting to use any other datamodel different from these ones will cause a "Unknown data model" ERROR in log traces. In addition, note that default datamodel for STH/Mongo sink is dm-by-entity as all others sinks which uses datamodel.
```
Comment (fgalan) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_mongo_sink.md:299]:
```suggestion
Note that attempting to use any other datamodel different from these ones will cause a "Unknown data model" ERROR in log traces. In addition, note that default datamodel for NGSIMongoSink is dm-by-entity as all others sinks which uses datamodel.
```

**Review by fgalan — APPROVED** [review]
LGTM

**Pull request merged**